shell/src/main/java/alluxio/shell/command/ChownCommand.java

```diff
@@ -61,8 +63,18 @@ public Options getOptions() {
     return new Options().addOption(RECURSIVE_OPTION);
   }
 
+  /*
+  See https://paulgorman.org/technical/presentations/linux_username_conventions.pdf
+  in which the author refers to IEEE Std 1003.1-2001 regarding the standards for
+  valid POSIX usernames.
+   */
+  private static final Pattern USER_PATTERN = Pattern.compile("(?<user>[\\w][\\w-]*[$]?)");
```

Comment:
`$` needs to be escaped. it should be `[\$]`?
Here is a good place to play around regexp 
http://www.regexplanet.com/advanced/java/index.html